lib/database/achievement.php

```diff
@@ -413,6 +413,12 @@ function AddEarnedAchievementJSON( $user, $achIDToAward, $isHardcore, $validatio
             $retVal[ 'Success' ] = FALSE;
             $retVal[ 'Error' ] = "User data cannot be found for $user";
         }
+        else if( $achData[ 'Flags' ] == 5 ) // Unofficial achievement
+        {
+            // should we return TRUE? does it affect something in the emulator's end?
+            $retVal[ 'Success' ] = FALSE;
+            $retVal[ 'Error' ] = "Unofficial achievements aren't registered on the RetroAchievements.org database";
```

Comment:
The emulator only looks for 'Error', not 'Success'. So at least right now it doesn't matter.

However, I would recommend leaving it as-is. Setting 'Success'='true' implies that the other non-error fields are populated.